splash/tests/mockserver.py

```diff
@@ -901,6 +901,22 @@ def render_GET(self, request):
             return base64.decodebytes(b'R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs=')
 
 
+class SubresourcesWithCaching(Subresources):
+    """ Embedded css and image """
+
+    @use_chunked_encoding
+    def render_GET(self, request):
+        return ("""<html><head>
+                <link rel="stylesheet" href="subresources-with-caching/style.css" />
```

Comment:
I think the problem could be that related resources don't set any specific cache headers, so the browser doesn't cache them. Could you try setting [Cache-Control](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#Caching_static_assets) header for these static resources?

Reply:
@kmike you are correct, I've pushed a new commit that sets that Cache-Control header and now it seems that at least for the image.gif resource, the entire scenario is reproducible.